tests/tools/kopiarunner/kopia_snapshotter.go

```diff
@@ -227,3 +270,109 @@ func parseManifestListForSnapshotIDs(output string) []string {
+
+// waitUntilServerStarted returns error if the Kopia API server fails to start before timeout.
+func (ks *KopiaSnapshotter) waitUntilServerStarted(ctx context.Context, addr string, serverStatusArgs ...string) error {
+	statusArgs := append([]string{"server", "status", "--address", addr}, serverStatusArgs...)
+
+	if err := retry.PeriodicallyNoValue(ctx, retryInterval, retryCount, waitingForServerString, func() error {
+		_, _, err := ks.Runner.Run(statusArgs...)
+		return err
+	}, retry.Always); err != nil {
+		return errors.New("server failed to start")
+	}
+
+	return nil
+}
+
+// createAndConnectServer creates Repository and a TLS server/client model for interaction.
+func (ks *KopiaSnapshotter) createAndConnectServer(serverAddr string, args ...string) (*exec.Cmd, error) {
+	if err := ks.ConnectOrCreateRepo(args...); err != nil {
+		return nil, err
+	}
+
+	var tempDir string
+
+	var tempDirErr error
+
+	if tempDir, tempDirErr = ioutil.TempDir("", "kopia"); tempDirErr != nil {
+		return nil, tempDirErr
+	}
+
+	defer os.RemoveAll(tempDir) // nolint:errcheck
+
+	tlsCertFile := filepath.Join(tempDir, "kopiaserver.cert")
+	tlsKeyFile := filepath.Join(tempDir, "kopiaserver.key")
+
+	serverArgs := []string{"--tls-generate-cert", "--tls-cert-file", tlsCertFile, "--tls-key-file", tlsKeyFile}
+
+	var cmd *exec.Cmd
+
+	var cmdErr error
+
+	if cmd, cmdErr = ks.CreateServer(serverAddr, serverArgs...); cmdErr != nil {
+		return nil, cmdErr
+	}
+
+	if err := certKeyExist(context.TODO(), tlsCertFile, tlsKeyFile); err != nil {
+		return nil, err
+	}
+
+	var fingerprint string
+
+	var fingerprintError error
+
+	if fingerprint, fingerprintError = getFingerPrintFromCert(tlsCertFile); fingerprintError != nil {
+		return nil, fingerprintError
+	}
+
+	serverAddr = fmt.Sprintf("https://%v", serverAddr)
+
+	if err := ks.waitUntilServerStarted(context.TODO(), serverAddr, "--server-cert-fingerprint", fingerprint); err != nil {
+		return cmd, err
+	}
+
+	clientArgs := []string{"--server-cert-fingerprint", fingerprint}
+	if err := ks.ConnectServer(serverAddr, clientArgs...); err != nil {
+		return nil, err
+	}
+
+	return cmd, nil
+}
+
+func getFingerPrintFromCert(path string) (string, error) {
+	pemData, err := ioutil.ReadFile(path) //nolint:gosec
+	if err != nil {
+		return "", err
+	}
+
+	block, rest := pem.Decode([]byte(pemData)) //nolint:unconvert
+	if block == nil || len(rest) > 0 {
+		return "", errors.New("pem decoding error")
+	}
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return "", err
+	}
+
+	fingerprint := sha256.Sum256(cert.Raw)
+
+	return hex.EncodeToString(fingerprint[:]), nil
+}
+
+func certKeyExist(ctx context.Context, tlsCertFile, tlsKeyFile string) error {
+	if err := retry.PeriodicallyNoValue(ctx, retryInterval, retryCount, "waiting for server to start", func() error {
+		if _, err := os.Stat(tlsCertFile); os.IsNotExist(err) {
+			return err
+		}
+		if _, err := os.Stat(tlsKeyFile); os.IsNotExist(err) {
+			return err
+		}
```

Comment:
IINM, one of the linters will fail here:
```suggestion
		}

		if _, err := os.Stat(tlsKeyFile); os.IsNotExist(err) {
			return err
		}

```

You can run the linter locally with `make lint`

Reply:
Ran `make lint`, its just fails with `Pdeathsig: syscall.SIGTERM` field not found.

Reply:
What OS?

Reply:
macOS, seems like this code will only work for Linux. What should we do?

Reply:
you could run it in a container, e.g. go container